platforms/chibios/drivers/serial_protocol.h

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 Stefan Kerkmann
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+
+#pragma once
+
+/**
+ * @brief Clears any intermediate sending or receiving state of the driver to a known good
+ * state. This happens after errors in the middle of transactions, to start with
+ * a clean slate.
+ */
+void driver_clear(void);
+
+/**
+ * @brief Driver specific initialization on the slave half.
+ */
+void driver_slave_init(void);
+
+/**
+ * @brief Driver specific specific initialization on the master half.
+ */
+void driver_master_init(void);
+
+/**
+ * @brief  Blocking receive of size * bytes.
+ *
+ * @return true Receive success.
+ * @return false Receive failed, e.g. by bit errors.
+ */
+bool __attribute__((nonnull, hot)) receive(uint8_t* destination, const size_t size);
+
+/**
+ * @brief Blocking receive of size * bytes with an implicitly defined timeout.
+ *
+ * @return true Receive success.
+ * @return false Receive failed, e.g. by timeout or bit errors.
+ */
+bool __attribute__((nonnull, hot)) receive_blocking(uint8_t* destination, const size_t size);
+
+/**
+ * @brief Blocking send of buffer with timeout.
+ *
+ * @return true Send success.
+ * @return false Send failed, e.g. by timeout or bit errors.
+ */
+bool __attribute__((nonnull, hot)) send(const uint8_t* source, const size_t size);
```

Comment:
Kiiiiiinda allergic to generic names like these.
What about:
- `receive` => `serial_transport_receive`
- `receive_blocking` => `serial_transport_receive_blocking`
- `send` => `serial_transport_send`

......just don't get me started about the naming of _serial_.

Reply:
Ah yes, forgot about C-style namespacing. Will apply that

Reply:
Done, applied the prefix to all public methods that are part of the protocol.